### ISSUE TITLE
Button accessible name context

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#994]
Back in 2016 we deferred resolution button context in https://github.com/w3c/wcag/issues/168 to WCAG Next..

If we had a piece of text such as a Person's name and then an Add button to the right do people agree that this is a failure of SC 1.3.1 or a pass?  the add button itself might say "add" but doesn't say what it adds.  Visually there is a relationship but not  programmatically.    2.4.6 only seems to address the label "add" -- which seems like it is clear visually in combination -- but it's the accessible name that is more problematic.    Adding "Add Persons's name" to the button would add extra text and then it would be best to either merge the button and text into one control or hide the text from screen readers in order to prevent extra verbosity.    

The authors of WCAG 2 felt that 2.4.4 was needed while 1.3.1 could have solved the issue of link context-- so does that mean button context is not covered by 1.3.1 or 2.4.6 either?   Buttons generally aren't in the same sentence or paragraph like links can be and button's may apply to groups of controls like a "submit" button that actually submits all fields in a form rather than just the field before it.  So where do we draw the lines for buttons?

**Comment by patrickhlauke** [comment]
> do people agree that this is a failure of SC 1.3.1 or a pass

i've begrudgingly marked these sorts of things as passes, BUT then usually give best practice advice (when there's a lot of similar controls all labelled the same) about disambiguating those buttons somehow (whether it be with visually hidden text, a more explicit `aria-label`, some `aria-describedby` relationship or similar). but i've seen this (and called it out many times) as a gap/loophole with WCAG 2.x. the weird mismatch of having 2.4.4 and 2.4.9 only for links, while not having anything comparable for any other types of controls.

i can see the argument for calling this out under 2.4.6, but arguably the presence of the "Person's name" text outside/next to the "Add" button would count as a form of label (as 2.4.6 doesn't really care about whether it's correctly associated in markup or not). though then yes, this does weigh the scales back in favour of perhaps dinging it under 1.3.1.

(and, as discussed in the other thread, i'm not convinced this could be dinged under 4.1.2 for the "name" not being descriptive/correct enough, as my reading of 4.1.2 doesn't say anything about the quality of the name, only that a control should have one...but again, that's debatable).

would be nice if there was a matching set of "control purpose (in context)" / "control purpose (control only)" type SCs (which could then supercede the "link purpose (in context)" / "link purpose (link only)", as links would fall under the more general "control" definition). this would also arguably make the new SCs more robust for applicability to things like native applications.
